### ISSUE TITLE
Cannot Create New Wasm Pack Project

### ISSUE THREAD

**Issue by seanaye** [rustwasm/wasm-pack#1266]
## 🐛 Bug description
`wasm-pack --version`
`> wasm-pack 0.11.0`

`wasm-pack new hello-world`
```
>Error: expected value at line 1 column 1
>Caused by: expected value at line 1 column 1
```

#### 🤔 Expected Behavior
Should create new project in current directory

#### 👟 Steps to reproduce
I tried to update wasm-pack via cargo 
`cargo install wasm-pack`
realized the install was broken, then uninstalled and re-installed via the install script
`cargo uninstall wasm-pack`
`curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh`
This gave me the above error

#### 🌍 Your environment
Include the relevant details of your environment.
wasm-pack version: see above
rustc version: rustc 1.69.0-nightly (246eae2fa 2023-02-21)

**Comment by wolkenmachine** [comment]
I'm hitting the same issue. Updated wasm-pack using:
`curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh`

error:
```
> wasm-pack new hello-world
Error: expected value at line 1 column 1
Caused by: expected value at line 1 column 1
```

**Comment by unti1x** [comment]
Confirm. `wasm-pack` v0.11.0 doesn't work, downgrading to 0.10.3 helped

**Comment by Dueen** [comment]
Issue confirmed: Creating new project not working on M1 Macbook.

**Comment by alikulaOwen** [comment]
The fix doesn't work for my Macbook pro m2 any other fixes?

**Comment by LeviticusNelson** [comment]
It might have to do with a bad get request to `https://crates.io/api/v1/crates/{name}` at `Krate::new`:
```
We require that all requests include a `User-Agent` header.  To allow us to determine the impact your bot has on our service, we ask that your user agent actually identify your bot, and not just report the HTTP client library you're using.  Including contact information will also reduce the chance that we will need to take action against your bot.

Bad:
  User-Agent: reqwest/0.9.1

Better:
  User-Agent: my_crawler

Best:
  User-Agent: my_crawler (my_crawler.com/info)
  User-Agent: my_crawler (help@my_crawler.com)

If you believe you've received this message in error, please email help@crates.io and include the request id 53cdcda1-0ea7-4acd-9673-45da30071f07.

```

**Comment by LeviticusNelson** [comment]
> It might have to do with a bad get request to `https://crates.io/api/v1/crates/cargo-generate` at `Krate::new`:
> 
> ```
> We require that all requests include a `User-Agent` header.  To allow us to determine the impact your bot has on our service, we ask that your user agent actually identify your bot, and not just report the HTTP client library you're using.  Including contact information will also reduce the chance that we will need to take action against your bot.
> 
> Bad:
>   User-Agent: reqwest/0.9.1
> 
> Better:
>   User-Agent: my_crawler
> 
> Best:
>   User-Agent: my_crawler (my_crawler.com/info)
>   User-Agent: my_crawler (help@my_crawler.com)
> 
> If you believe you've received this message in error, please email help@crates.io and include the request id 53cdcda1-0ea7-4acd-9673-45da30071f07.
> ```

I added a User-Agent header and now I get this as an error:
```
dyld[77274]: Library not loaded: /usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib
  Referenced from: <9670E1DF-008C-328F-B608-CDADE7311C0A> /Users/levi/.cargo/bin/cargo-generate
  Reason: tried: '/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (no such file), '/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/Users/levi/wasm-pack/target/debug/build/bzip2-sys-e4c1951e741fbbdc/out/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/openssl-sys-ec0a95688bf13e99/out/openssl-build/install/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/zstd-sys-dbc8793556ebaa34/out/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/deps/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/rust/1.69.0/lib/rustlib/aarch64-apple-darwin/lib/libssl.1.1.dylib' (no such file), '/Users/levi/lib/libssl.1.1.dylib' (no such file), '/usr/local/lib/libssl.1.1.dylib' (no such file), '/usr/lib/libssl.1.1.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/Users/levi/wasm-pack/target/debug/build/bzip2-sys-e4c1951e741fbbdc/out/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/openssl-sys-ec0a95688bf13e99/out/openssl-build/install/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/zstd-sys-dbc8793556ebaa34/out/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/deps/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/rust/1.69.0/lib/rustlib/aarch64-apple-darwin/lib/libssl.1.1.dylib' (no such file), '/Users/levi/lib/libssl.1.1.dylib' (no such file), '/usr/local/lib/libssl.1.1.dylib' (no such file), '/usr/lib/libssl.1.1.dylib' (no such file, not in dyld cache)
Error: failed to execute `cargo-generate`: exited with signal: 6 (SIGABRT)
  full command: "/Users/levi/.cargo/bin/cargo-generate" "--version"
Caused by: failed to execute `cargo-generate`: exited with signal: 6 (SIGABRT)
  full command: "/Users/levi/.cargo/bin/cargo-generate" "--version"
```

**Comment by LeviticusNelson** [comment]
> > It might have to do with a bad get request to `https://crates.io/api/v1/crates/cargo-generate` at `Krate::new`:
> > ```
> > We require that all requests include a `User-Agent` header.  To allow us to determine the impact your bot has on our service, we ask that your user agent actually identify your bot, and not just report the HTTP client library you're using.  Including contact information will also reduce the chance that we will need to take action against your bot.
> > 
> > Bad:
> >   User-Agent: reqwest/0.9.1
> > 
> > Better:
> >   User-Agent: my_crawler
> > 
> > Best:
> >   User-Agent: my_crawler (my_crawler.com/info)
> >   User-Agent: my_crawler (help@my_crawler.com)
> > 
> > If you believe you've received this message in error, please email help@crates.io and include the request id 53cdcda1-0ea7-4acd-9673-45da30071f07.
> > ```
> 
> I added a User-Agent header and now I get this as an error:
> 
> ```
> dyld[77274]: Library not loaded: /usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib
>   Referenced from: <9670E1DF-008C-328F-B608-CDADE7311C0A> /Users/levi/.cargo/bin/cargo-generate
>   Reason: tried: '/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (no such file), '/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/Users/levi/wasm-pack/target/debug/build/bzip2-sys-e4c1951e741fbbdc/out/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/openssl-sys-ec0a95688bf13e99/out/openssl-build/install/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/zstd-sys-dbc8793556ebaa34/out/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/deps/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/rust/1.69.0/lib/rustlib/aarch64-apple-darwin/lib/libssl.1.1.dylib' (no such file), '/Users/levi/lib/libssl.1.1.dylib' (no such file), '/usr/local/lib/libssl.1.1.dylib' (no such file), '/usr/lib/libssl.1.1.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/Users/levi/wasm-pack/target/debug/build/bzip2-sys-e4c1951e741fbbdc/out/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/openssl-sys-ec0a95688bf13e99/out/openssl-build/install/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/zstd-sys-dbc8793556ebaa34/out/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/deps/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/rust/1.69.0/lib/rustlib/aarch64-apple-darwin/lib/libssl.1.1.dylib' (no such file), '/Users/levi/lib/libssl.1.1.dylib' (no such file), '/usr/local/lib/libssl.1.1.dylib' (no such file), '/usr/lib/libssl.1.1.dylib' (no such file, not in dyld cache)
> Error: failed to execute `cargo-generate`: exited with signal: 6 (SIGABRT)
>   full command: "/Users/levi/.cargo/bin/cargo-generate" "--version"
> Caused by: failed to execute `cargo-generate`: exited with signal: 6 (SIGABRT)
>   full command: "/Users/levi/.cargo/bin/cargo-generate" "--version"
> ```

Update: I reinstalled cargo-generate -> `cargo uninstall cargo-generate && cargo install cargo-generate` and it works fine now. Still needed USER-AGENT, so I can create a PR

**Comment by drager** [comment]
Hmm, I haven't seen this issue myself... Does it work fine in `0.11.1`?

**Comment by LeviticusNelson** [comment]
When I added the header for `0.11.1`, it worked, but was giving me errors when there was no USER-AGENT header.

**Comment by ben-xD** [comment]
I've just tested it on `0.11.1`, and I get 
```
$ wasm-pack new hello-wasm
Error: expected value at line 1 column 1
Caused by: expected value at line 1 column 1
```
## version
```
$ wasm-pack --version
wasm-pack 0.11.1
```

cc: @orhun since he said 0.11.1 fixes it in https://github.com/rustwasm/wasm-pack/issues/1245#issuecomment-1546147124

**Comment by orhun** [comment]
Yeah, it is still a mystery to me how it stopped working and start working again. Somebody need to dive deeper into this.

**Comment by drager** [comment]
I just merged #1285. Please try out latest master and see if that works for you @ben-xD @orhun

**Comment by clbarnes** [comment]
Could releasing this fix be prioritised, if only as a patch release on top of the previous release without any other recent additions? Older wasm-pack versions raise a warning if an older version is used so at present we're just funnelling users into a completely broken state.

**Comment by drager** [comment]
> Could releasing this fix be prioritised, if only as a patch release on top of the previous release without any other recent additions? Older wasm-pack versions raise a warning if an older version is used so at present we're just funnelling users into a completely broken state.

Released in v0.12.0
* https://github.com/rustwasm/wasm-pack/releases/tag/v0.12.0
* https://crates.io/crates/wasm-pack/0.12.0
* https://www.npmjs.com/package/wasm-pack/v/0.12.0

**Comment by ddimaria** [comment]
> > > It might have to do with a bad get request to `https://crates.io/api/v1/crates/cargo-generate` at `Krate::new`:
> > > ```
> > > We require that all requests include a `User-Agent` header.  To allow us to determine the impact your bot has on our service, we ask that your user agent actually identify your bot, and not just report the HTTP client library you're using.  Including contact information will also reduce the chance that we will need to take action against your bot.
> > > 
> > > Bad:
> > >   User-Agent: reqwest/0.9.1
> > > 
> > > Better:
> > >   User-Agent: my_crawler
> > > 
> > > Best:
> > >   User-Agent: my_crawler (my_crawler.com/info)
> > >   User-Agent: my_crawler (help@my_crawler.com)
> > > 
> > > If you believe you've received this message in error, please email help@crates.io and include the request id 53cdcda1-0ea7-4acd-9673-45da30071f07.
> > > ```
> > 
> > 
> > I added a User-Agent header and now I get this as an error:
> > ```
> > dyld[77274]: Library not loaded: /usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib
> >   Referenced from: <9670E1DF-008C-328F-B608-CDADE7311C0A> /Users/levi/.cargo/bin/cargo-generate
> >   Reason: tried: '/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (no such file), '/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/Users/levi/wasm-pack/target/debug/build/bzip2-sys-e4c1951e741fbbdc/out/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/openssl-sys-ec0a95688bf13e99/out/openssl-build/install/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/zstd-sys-dbc8793556ebaa34/out/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/deps/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/rust/1.69.0/lib/rustlib/aarch64-apple-darwin/lib/libssl.1.1.dylib' (no such file), '/Users/levi/lib/libssl.1.1.dylib' (no such file), '/usr/local/lib/libssl.1.1.dylib' (no such file), '/usr/lib/libssl.1.1.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/openssl@1.1/1.1.1t/lib/libssl.1.1.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/Users/levi/wasm-pack/target/debug/build/bzip2-sys-e4c1951e741fbbdc/out/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/openssl-sys-ec0a95688bf13e99/out/openssl-build/install/lib/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/build/zstd-sys-dbc8793556ebaa34/out/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/deps/libssl.1.1.dylib' (no such file), '/Users/levi/wasm-pack/target/debug/libssl.1.1.dylib' (no such file), '/opt/homebrew/Cellar/rust/1.69.0/lib/rustlib/aarch64-apple-darwin/lib/libssl.1.1.dylib' (no such file), '/Users/levi/lib/libssl.1.1.dylib' (no such file), '/usr/local/lib/libssl.1.1.dylib' (no such file), '/usr/lib/libssl.1.1.dylib' (no such file, not in dyld cache)
> > Error: failed to execute `cargo-generate`: exited with signal: 6 (SIGABRT)
> >   full command: "/Users/levi/.cargo/bin/cargo-generate" "--version"
> > Caused by: failed to execute `cargo-generate`: exited with signal: 6 (SIGABRT)
> >   full command: "/Users/levi/.cargo/bin/cargo-generate" "--version"
> > ```
> 
> Update: I reinstalled cargo-generate -> `cargo uninstall cargo-generate && cargo install cargo-generate` and it works fine now. Still needed USER-AGENT, so I can create a PR

I had this same issue and running `cargo uninstall cargo-generate && cargo install cargo-generate` fixed the issue for me as well.
